www/src/Pages/InputMacroAddonPage.tsx

```diff
@@ -98,13 +98,10 @@ const FormContext = () => {
 		}
 		fetchData();
 	}, [setValues]);
-/*
+
 	useEffect(() => {
-		async function setData() {
-			await setValues(filterMacroInputs(values));
-		}
-		setData();
-	}, [values, setValues]);*/
+		setValues(filterMacroInputs(values));
```

Comment:
this will crash your browser if you type in the name of the macro too fast.

I think this is a React bug, so we just need to implement some easy rate-limit:

`if ( time_since_update < rate_limit ) {
       updateValues = values;
       setTimeout(); wake-up
} else {
       setValues(filteredMacroInputs(updateValues));
}`
or something

Reply:
I refactored the existing logic to pass the form update handler to the child component instead of trying to rely on useEffect in the child component. This allowed the removal of the EMPTY_INPUT logic on the page, and was replaced with a real delete (splice) of the input from the array.